### ISSUE TITLE
How to downgrade a database

### ISSUE THREAD

**Issue by freeChange** [Raizlabs/DBFlow#1164]
DBFlow Version:3.1.0
Issue Kind (Bug, Question, Feature):Question

Description: 
How to do database downgrade processing?

**Comment by joaopmafra** [comment]
Hi,

Any answer to this problem? I have an existing application that uses DBFlow and I need database downgrade. If there is no solution, I will have to remove DBFlow from the project and change several lines of code ;(

**Comment by agrosner** [comment]
Im curious, why do you need to downgrade your db?

**Comment by agrosner** [comment]
what kind of operation are you reversing or attempting to?

**Comment by joaopmafra** [comment]
@agrosner 

First of all, thanks for your response. 

I have a client that distributes the app outside of google play. He wants to distribute a version of the app and have a way to downgrade the app if their users don't like that version. Reinstalling the app can lead to data loss because users can use the new version and then want to go back to the old one. My only option is downgrade the database.

**Comment by agrosner** [comment]
what kind of operation on the db do you need to do? also to note that SQLite does not support all features. It only supports adding columns, dropping/adding tables, renaming tables, and I think that's it for modifying the structure. So depending on what you change, you actually might not be able to roll it back.

**Comment by joaopmafra** [comment]
The operations you mentioned are enough for me and I aware of the risks. 

Also I need to say that I found workaround to the problem. I simply created a subclass of SQLiteOpenHelper for my database and put the code for creating and migrating the database in it. In the application class, I open the database using the SQLiteOpenHelper subclass, close it and after that I do the DBflow normal initialization proccess. For me it's working well, I don't know if I need that feature implemented in DBFlow anymore.
